File: vika_3/planetaris.py
import sys

s = []
for i in sys.stdin:
    temp = list(map(lambda c: int(c), i.split()))
    s.append(temp)

def attack(ships, fleet, points):
    if ships < 0:
        return -1
    elif len(fleet) == 0:
        return 0

    left = attack(ships, fleet[1:], points)
    right = attack(ships - (fleet[0] + 1), fleet[1:], points)

    if right >= left:
        return points + 1 + right
    return points + left

print(attack(s[0][1], s[1], 0))

File: vika_3/planetaris.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let lines: Vec<String> = stdin
        .lock()
        .lines()
        .map(|l| l.unwrap())
        .collect::<Vec<String>>();

    let header = lines[0]
        .split(" ")
        .map(|i| i.parse::<i32>().unwrap())
        .collect::<Vec<i32>>();

    let (n, mut ships) = (header[0], header[1]);

    let mut atli = lines[1]
        .split(" ")
        .map(|w| w.parse::<i32>().unwrap())
        .collect::<Vec<i32>>();

    atli.sort();

    let mut wins = 0;
    for fleet in atli {}

    println!("{wins}");
}
